### DISCUSSION TITLE
Help me understand why Akamai EdgeDNS challenge provider failed.

### DISCUSSION THREAD

**Discussion by lystopad** [go-acme/lego#1904] [Q&A] [answered]
Hello all!

I am using terraform provider "vancluever/acme/latest" with terraform for automation of creation of Signed SSL Certificates.

Recent jenkins build failed (without any changes in terraform code) with following error:

```
16:36:49  [*.mydomain.local] [*.mydomain.local] acme: error presenting token: 2 errors occurred:
16:36:49  	* rpc error: code = Unknown desc = edgedns: Record not found.
16:36:49  	* error encountered while presenting token for DNS challenge: rpc error: code = Unknown desc = edgedns: Record not found.
```

I tried to debug terraform. No luck. I tried to destroy and deploy it again. No luck.

This error code is not clear. I have no idea how to fix it.

I tried to play with different parameters like `AKAMAI_POLLING_INTERVAL` or `AKAMAI_PROPAGATION_TIMEOUT` without success.

Could you, please, advise how to troubleshoot it ?
Is there any ideas to change error code to something more clear which will help fix it?

Thanks.

Best regards
// lystopad

**Comment by ldez** [comment]
Hello,

Have you updated lego?

**Reply by lystopad** [reply]
Thanks for you reply @ldez .

As I wrote -- I am using "vancluever/acme/latest".
Just checked -- it is is used `lego` v4.10.0.

**Reply by ldez** [reply]
I asked to know if there is a variation of versions.

The error is not related to `AKAMAI_POLLING_INTERVAL` or `AKAMAI_PROPAGATION_TIMEOUT` because it's an error from the EdgeDNS API.

You may have a CNAME or a wildcard CNAME.

So can you try to set the env var `LEGO_DISABLE_CNAME_SUPPORT` to `true`?

**Reply by lystopad** [reply]
Thank you Idez.

I tried it as env.variable for terraform. But it does not help.

Do you have any other ideas? Should I try some legacy version instead?

thanks.

**Reply by lystopad** [reply, marked as answer]
I hardcoded old terraform provider version and it helped me.

```
        acme = {
            source = "vancluever/acme"
            version = "2.12.0"
        }
```
